flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/RowTypeTest.scala

```diff
@@ -142,14 +142,11 @@ class RowTypeTest extends RowTypeTestBase {
 
   @Test
   def testUnsupportedCastSqlApi(): Unit = {
-    expectedException.expect(classOf[CodeGenException])
-    expectedException.expectMessage(
-      "Unsupported cast from 'ROW<`f0` STRING, `f1` BOOLEAN>' to 'ROW<`f0` INT, `f1` STRING>'")
+    expectedException.expect(classOf[ValidationException])
```

Comment:
nit: Can we assert the message to make sure we are failing for the right reason?